.github/workflows/checks.yml

```diff
@@ -0,0 +1,63 @@
+name: CI
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/checks.yaml'
+      - '**.go'
+
+permissions:
+  # Permission for checking out code
+  contents: read
+
+jobs:
+  build:
+    name: Check & Build Provider
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: chmod +x ./scripts/gofmtcheck.sh
+      - name: gofmt Check
+        run: ./scripts/gofmtcheck.sh
+      # - name: Generate provider code
+      #   run: go generate
+      # - name: Check generated code for diffs
+      #   run: git diff --exit-code
+      - name: Build
+        run: go build -v
+  acceptance:
+    name: Acceptance Tests
+    needs: build
+    runs-on: ubuntu-latest
+    concurrency:
+        group: tf-aci-ci-test-${{ matrix.apic_host.name }}
+        cancel-in-progress: false
+    strategy:
+      fail-fast: false
+      matrix:
+        apic_host:
+          - name: v5.2
+            url: 'https://173.36.219.83/'
+          - name: v6.0
+            url: 'https://173.36.219.84/'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: '1.7.*'
+          terraform_wrapper: false
+      - name: Terraform Acceptance Test (APIC ${{ matrix.apic_host.name }})
+        run: go test github.com/CiscoDevNet/terraform-provider-aci/v2/internal/provider -v -cover -timeout 120m

```

Comment:
Can we get the coverage output saved somewhere? (not sure if we can also use codecov.io for golang)

Reply:
Good idea. I will look into it. 👍

Reply:
I added the CodeCov step but it looks like we need to provide a token via a secret. However, I don't have admin access to do that myself. But it should run successfully when the CodeCov token secret is configured as an environment variable.